src/wasm-traversal.h

```diff
@@ -176,17 +176,33 @@ struct Walker : public VisitorType {
     setFunction(func);
     static_cast<SubType*>(this)->doWalkFunction(func);
     static_cast<SubType*>(this)->visitFunction(func);
+    setFunction(nullptr);
```

Comment:
Yeah, we init it to `nullptr`. So with this change it should always be in the right state.